Swiften/MIX/MIXRegistry.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2017 Tarun Gupta
+ * Licensed under the simplified BSD license.
+ * See Documentation/Licenses/BSD-simplified.txt for more information.
+ */
+
+#pragma once
+
+#include <unordered_set>
+
+#include <boost/signals2.hpp>
+
+#include <Swiften/Base/API.h>
+#include <Swiften/JID/JID.h>
+#include <Swiften/MIX/MIX.h>
+#include <Swiften/MIX/MIXImpl.h>
+#include <Swiften/Elements/MIXJoin.h>
+#include <Swiften/Elements/MIXLeave.h>
+#include <Swiften/Elements/ErrorPayload.h>
+#include <Swiften/Roster/XMPPRoster.h>
+#include <Swiften/Queries/IQRouter.h>
+#include <Swiften/Queries/GenericRequest.h>
+
+namespace Swift {
+
+    class SWIFTEN_API MIXRegistry {
+        public:
+            MIXRegistry(const JID& ownJID, IQRouter* iqRouter, XMPPRoster* xmppRoster);
+
+            ~MIXRegistry();
+
+            void joinChannel(const JID& channelJID, const std::unordered_set<std::string>& nodes);
+
+            void leaveChannel(const JID& channelJID);
```

Comment:
I'd say if registry is handling the join and keeping track of joined channels, leave should also be here as successfully leaving a channel depends on (onJIDRemoved).